### PR TITLE
chore: adding `Build and deploy VC-API` workflow

### PR DESCRIPTION
**Pull request by artursudnik**

### CODE REVIEW

File: .github/workflows/deploy-vc-api.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: Build and deploy VC-API
+
+on: [push, workflow_dispatch]
```
Comment (artursudnik) [inline .github/workflows/deploy-vc-api.yml]: this needs to be adjusted before merging

File: .github/workflows/deploy-vc-api.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+      - name: Build Docker image
+        working-directory: ./vc-api-owf
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr-ssi.outputs.registry }}
+          AWS_ECR_REPOSITORY_SSI: ${{ secrets.DEV_ECR_REPOSITORY }}
+        run: |
+          echo $ECR_REGISTRY
+          docker build \
+            -t $ECR_REGISTRY/$AWS_ECR_REPOSITORY_SSI \
```
Comment (artursudnik) [inline .github/workflows/deploy-vc-api.yml]: a tag to be added

File: .github/workflows/deploy-vc-api.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            -f apps/vc-api/Dockerfile .
+
+#      - name: Push Docker image
+#        working-directory: ./vc-api-owf
+#        run: |
+#          docker push \
+#            $ECR_REGISTRY/$AWS_ECR_REPOSITORY_SSI
```
Comment (artursudnik) [inline .github/workflows/deploy-vc-api.yml]: a tag to be added

Comment (jrhender) [inline .github/workflows/deploy-vc-api.yml]: Is this just commented for now while testing the other steps?

**Review by jrhender — COMMENTED** [review]
I think the workflow is missing the two steps below (which are currently in the `deploy` workflow). Are you thinking to add them later?
```
      - name: Deploy SSI
        uses: clowdhaus/argo-cd-action/@v1.9.0
        id: argocd_login_ssi
        with:
          command: login ${{ steps.env_vars.outputs.argocd_url }}
          options: --insecure --password ${{ steps.env_vars.outputs.argocd_password }} --username ${{ steps.env_vars.outputs.argocd_username }}

      - name: ArgoCD overvrite SSI values.yaml
        uses: clowdhaus/argo-cd-action/@v1.9.0
        id: argocd_image_tag_overwrite_ssi
        with:
          command: app set ssi-wallet
          options: -p image.tag=${{ steps.tag_version.outputs.new_tag }}
```

File: .github/workflows/deploy-vc-api.yml (hunk 3 above)
Comment (artursudnik) [reply .github/workflows/deploy-vc-api.yml]: Yes, I just don't want image to be pushed to the repository when testing this.

**Comment by artursudnik** [comment]
> I think the workflow is missing the two steps below (which are currently in the `deploy` workflow). Are you thinking to add them later?
> 
> ```
>       - name: Deploy SSI
>         uses: clowdhaus/argo-cd-action/@v1.9.0
>         id: argocd_login_ssi
>         with:
>           command: login ${{ steps.env_vars.outputs.argocd_url }}
>           options: --insecure --password ${{ steps.env_vars.outputs.argocd_password }} --username ${{ steps.env_vars.outputs.argocd_username }}
> 
>       - name: ArgoCD overvrite SSI values.yaml
>         uses: clowdhaus/argo-cd-action/@v1.9.0
>         id: argocd_image_tag_overwrite_ssi
>         with:
>           command: app set ssi-wallet
>           options: -p image.tag=${{ steps.tag_version.outputs.new_tag }}
> ```

Steps are missing because it is not finished yet and to avoid triggering deployments when testing the workflow.

**Review by jrhender — APPROVED** [review]

**Pull request merged**